eclair-core/src/test/scala/fr/acinq/eclair/payment/Bolt12InvoiceSpec.scala

```diff
@@ -58,219 +58,123 @@ class Bolt12InvoiceSpec extends AnyFunSuite {
     val (nodeKey, payerKey, chain) = (randomKey(), randomKey(), randomBytes32())
     val offer = Offer(Some(10000 msat), "test offer", nodeKey.publicKey, Features.empty, chain)
     val request = InvoiceRequest(offer, 11000 msat, 1, Features.empty, payerKey, chain)
-    val invoice = Bolt12Invoice(offer, request, randomBytes32(), nodeKey, CltvExpiryDelta(20), Features.empty, Seq(createPaymentBlindedRoute(nodeKey.publicKey)))
-    assert(invoice.isValidFor(offer, request))
-    assert(invoice.checkSignature())
-    assert(!invoice.checkRefundSignature())
+    val invoice = Bolt12Invoice(request, randomBytes32(), nodeKey, 300 seconds, Features.empty, Seq(createPaymentBlindedRoute(nodeKey.publicKey)))
+    assert(invoice.isValidFor(request))
```

Comment:
In this test and the next one, we should use `invoice.checkSignature()`, not `invoice.isValidFor()`, the only part that should fail validation is the signature part so we should ignore the rest (or we can `assert` on both, but it's useful to have one explicitly on `checkSignature()`). Otherwise if anything else in `isValidFor` makes that function return `false`, we end up not testing the signature code.